### ISSUE TITLE
[Alert] - use single auto-close speed

### ISSUE THREAD

**Issue by macandcheese** [Esri/calcite-design-system#6395]
**monday.com sync:** #18005596673

### Description

Initially brought up as a comment on https://github.com/Esri/calcite-components/issues/6363, we should consider a single, recommended auto-close duration and not have a configuration option. Fast is too fast and can be frustrating for users. 

Defer to @ashetland @SkyeSeitz on the appropriate value for this.

### Acceptance Criteria

A single, recommended, usable speed is used for ‘auto-close’ to reduce fragmentation.

**Update:** per https://github.com/Esri/calcite-design-system/issues/6395#issuecomment-2087128484,

* use `medium`'s value as the single, default speed.
* update `autoCloseDuration` to take a number as the custom duration
* deprecate specifying scale-based values
* create follow-up issue to drop scale-based value support in the next breaking-change release after 5.0.

### Which Component

Alert

### Esri team

Calcite (design)

**Comment by ashetland** [comment]
It'll likely require some testing to get this one right. At first glance the current medium speed isn't too bad, but the fast speed, in hindsight, maybe isn't terrible either. So maybe somewhere in between would work best.

The visual of the decreasing bar adds a sense of urgency that I don't find appealing, if I'm honest. It makes all the speeds _feel_ faster than they actually are. It might be worth exploring an alternative way to represent that the Alert will be closing by itself. Just spitballing here, but if auto-closing Alerts had a place to go instead of disappearing entirely then the countdown would be less necessary — similar to notification/notification center patterns.

**Comment by macandcheese** [comment]
I do think it's important to maintain separation of intent between Alerts and something like a system-wide notification system and aggregation of such to a user-store. 

There's definitely possible overlap where, say, an alert that messages to a user a process has begun, has a reciprocal system-notification available, but for the time being that seems out of scope for the simple Alert component.

I'm not sure we've received much feedback on the progress bar itself (save for times when the alert's contents "processing begun" have been conflated with that process itself, but that likely isn't a great use case of an auto-closing alert for that reason). Auto closing alerts that dismiss without warning can be a frustration point for users if it dismisses right before they "reach it", so to speak. We could explore alternative displays though.

**Comment by geospatialem** [comment]
This also is a consideration as part of https://github.com/Esri/calcite-components/issues/6358, where `alert` has some pretty noticeable animations when animations are disabled/reduced on folks' OS.

As part of this effort, we should consider disabling `auto-close` for those with this feature enabled.

In looking at other design systems, wasn't able to spot a good example where `alert` supports a timed close **and** supports disabled/reduced animations.

**Comment by geospatialem** [comment]
Related https://github.com/Esri/calcite-design-system/issues/6793

**Comment by geospatialem** [comment]
Proceeding to use the `"medium"` speed and moving this to a future milestone for development efforts.

**Comment by ashetland** [comment]
> Proceeding to use the `"medium"` speed and moving this to a future milestone for development efforts.

Marking `ready for dev` based on this comment.

**Comment by macandcheese** [comment]
I think we can keep it but use it more like in Carousel - where it accepts a custom value - https://developers.arcgis.com/calcite-design-system/components/carousel/#api-reference-properties-autoplayDuration

**Comment by ashetland** [comment]
> I think we can keep it but use it more like in Carousel - where it accepts a custom value - https://developers.arcgis.com/calcite-design-system/components/carousel/#api-reference-properties-autoplayDuration

Yep, agreed. We can simplify things to one default speed, but still allow customization of that speed.

**Comment by geospatialem** [comment]
The first step of the scope above is to deprecate the 3 animation speeds with future removal ~~in 4.0~~ and support via a new property, matching `carousel`'s `duration` prop.

**Comment by isaacbraun** [comment]
The proposed `autoCloseDuration` would resolve the ask in #6793 and should solve [Matt's concern](https://github.com/Esri/calcite-design-system/issues/6793#issuecomment-2803393317) about a singular source of truth. Closing #6793 as duplicate of this issue.
